### PR TITLE
[training] add ds support to lora sd3.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Fixes https://github.com/huggingface/diffusers/pull/10252. 

Cc: @leisuzz could you give this a try?

### CODE REVIEW

**Comment by leisuzz** [comment]
@sayakpaul Please take a look at the comment, especially the third comment has to be changed!

**Comment by sayakpaul** [comment]
> Please take a look at the comment, especially the third comment has to be changed!

No idea what this means.

**Comment by leisuzz** [comment]
@sayakpaul In the line 1852, I added the modification on `if args.checkpoints_total_limit is not None:`
It should be `if accelerator.is_main_process and args.checkpoints_total_limit is not None:`. Otherwise errors will occur

**Comment by sayakpaul** [comment]
Can you comment on the changes directly instead? That will be helpful and easier.

**Review by leisuzz — CHANGES_REQUESTED** [review]

File: examples/dreambooth/train_dreambooth_lora_sd3.py [hunk 1]
```diff
@@ -1292,11 +1292,13 @@ def save_model_hook(models, weights, output_dir):
             text_encoder_two_lora_layers_to_save = None
 
             for model in models:
-                if isinstance(model, type(unwrap_model(transformer))):
+                if isinstance(unwrap_model(model), type(unwrap_model(transformer))):
+                    model = unwrap_model(model)
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply examples/dreambooth/train_dreambooth_lora_sd3.py:1296]: `else` should not be needed as the model would already be type-casted.

File: examples/dreambooth/train_dreambooth_lora_sd3.py [hunk 2]
```diff
@@ -1292,11 +1292,13 @@ def save_model_hook(models, weights, output_dir):
...
                     transformer_lora_layers_to_save = get_peft_model_state_dict(model)
-                elif isinstance(model, type(unwrap_model(text_encoder_one))):  # or text_encoder_two
+                elif isinstance(unwrap_model(model), type(unwrap_model(text_encoder_one))):  # or text_encoder_two
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply examples/dreambooth/train_dreambooth_lora_sd3.py]: Addressed

File: examples/dreambooth/train_dreambooth_lora_sd3.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sayakpaul) [reply examples/dreambooth/train_dreambooth_lora_sd3.py:1296]: Addressed

File: examples/dreambooth/train_dreambooth_lora_sd3.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (linoytsaban) [reply examples/dreambooth/train_dreambooth_lora_sd3.py]: isn't it already be reflected in the models sent to `save_model_hook`?

File: examples/dreambooth/train_dreambooth_lora_sd3.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (linoytsaban) [reply examples/dreambooth/train_dreambooth_lora_sd3.py:1296]: why is the change of `isinstance(model,...` to `isinstance(unwrap_model(model),...` is needed in the if statement?

**Review by linoytsaban — COMMENTED** [review]
thanks a lot @leisuzz! LGTM, just a couple of small comments

File: examples/dreambooth/train_dreambooth_lora_sd3.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sayakpaul) [reply examples/dreambooth/train_dreambooth_lora_sd3.py:1296]: That is a deepspeed-specific change as with deepspeed, the model gets wrapped into a `Module`.

File: examples/dreambooth/train_dreambooth_lora_sd3.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (linoytsaban) [reply examples/dreambooth/train_dreambooth_lora_sd3.py]: nvm addressed

**Review by linoytsaban — APPROVED** [review]

**Pull request merged**